src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -166,16 +167,11 @@ public bool Equals(LockFile other)
 
             return IsLocked == other.IsLocked
                 && Version == other.Version
-                && ProjectFileDependencyGroups.OrderBy(group => group.FrameworkName, StringComparer.OrdinalIgnoreCase)
-                    .SequenceEqual(other.ProjectFileDependencyGroups.OrderBy(
-                        group => group.FrameworkName, StringComparer.OrdinalIgnoreCase))
-                && Libraries.OrderBy(library => library.Name, StringComparer.OrdinalIgnoreCase)
-                    .SequenceEqual(other.Libraries.OrderBy(library => library.Name, StringComparer.OrdinalIgnoreCase))
-                && Targets.OrderBy(target => target.Name).SequenceEqual(other.Targets.OrderBy(target => target.Name))
-                && ProjectFileToolGroups.OrderBy(group => group.FrameworkName, StringComparer.OrdinalIgnoreCase)
-                    .SequenceEqual(other.ProjectFileToolGroups.OrderBy(
-                        group => group.FrameworkName, StringComparer.OrdinalIgnoreCase))
-                && Tools.OrderBy(target => target.Name).SequenceEqual(other.Tools.OrderBy(target => target.Name));
+                && ProjectFileDependencyGroups.OrderedEquals(other.ProjectFileDependencyGroups, group => group.FrameworkName, StringComparer.OrdinalIgnoreCase)
+                && Libraries.OrderedEquals(other.Libraries, library => library.Name, StringComparer.OrdinalIgnoreCase)
+                && Targets.OrderedEquals(other.Targets, target => target.Name)
```

Comment:
This should probably use StringComparer.Ordinal instead of the default culture sensitive one. For this reason I would also suggest making all parameters on the extension required instead of optional.